src/crimson/common/logfile.cc

```diff
@@ -0,0 +1,76 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include <fstream>
+#include <seastar/util/log.hh>
+
+#include "common/debug.h"
+#include "logfile.h"
+#include "log.h"
+
+#define dout_context g_ceph_context
+#define dout_prefix *_dout
+#define dout_subsys ceph_subsys_
+
+namespace {
+  seastar::logger& logger() {
+    return crimson::get_logger(ceph_subsys_osd);
+  }
+}  // namespace
+
+struct logfh {
+  std::string log_file_path;
+  std::ofstream log_ofstream;
+
+  ~logfh() {
+    log_ofstream.flush();
```

Comment:
i don't think we need to take care of this. `logger::do_log()` does call `_out->flush()` for every log entry.